src/avram/callbacks.cr

```diff
@@ -24,6 +24,18 @@ module Avram::Callbacks
     end
   end
 
+  macro before_run(method_name)
```

Comment:
This does the exact same thing as `before_save`, but it feels weird calling `before_save` in an Operation. To clean this up, we could just have `before_run` call `before_save` internally... but then if we ever needed `before_save` to act different, we'd end up having to break them out again.

Reply:
This is some tricky naming.  What about `before_execution`?

Reply:
The method that you call in these is called `run`. So like `LoginUser.run(params)`. This was actually sort of modeled after the mutations gem in Ruby 😅  (though, that doesn't have callbacks).

Reply:
I think I'd flip it. So `before_save` calls `before_run` under the hood. That way if `before_save` needs to change something we can modify just `before_save` and leave `before_run` alone